backend/oasst_backend/api/v1/messages.py

```diff
@@ -354,3 +354,21 @@ def emoji_tx(session: deps.Session):
         return pr.handle_message_emoji(message_id, request.op, request.emoji)
 
     return utils.prepare_message(emoji_tx())
+
+
+@router.get("/search", response_model=list[protocol.Message])
+def search_messages(
+    *,
+    query: str,
+    lang: str,
+    frontend_user: deps.FrontendUserId = Depends(deps.get_frontend_user_id),
+    api_client: ApiClient = Depends(deps.get_api_client),
+):
+    """Perform text search over messages in given language."""
+
+    @managed_tx_function(CommitMode.READONLY)
+    def search_tx(session: deps.Session) -> list[Message]:
+        pr = PromptRepository(session, api_client, frontend_user=frontend_user)
+        return pr.search_messages(query, lang)
+
+    return utils.prepare_message_list(search_tx())
```

Comment:
Can you merge this to the /cursor endpoint?

Reply:
Yeah I think that would be fine. I'll do some local testing to ensure this implementation works and then merge the endpoints when finished

Reply:
Merged